src/components/MissionRow.js

```diff
@@ -1,23 +1,28 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+import { joinMission } from '../redux/missions/missions';
 
 const MissionRow = (props) => {
-  const { name, description } = props;
+  const { name, description, id } = props;
+  const dispatch = useDispatch();
+
+  const handleClick = (e) => {
+    dispatch(joinMission(e.target.id));
+  };
 
   return (
     <tr>
       <th>{name}</th>
       <th>
-        <p>
-          {description}
-        </p>
+        <p>{description}</p>
         <br />
       </th>
       <th className="badge">
         <span>not a member</span>
       </th>
       <th className="join">
-        <button type="button">
+        <button onClick={(e) => handleClick(e)} id={id} type="button">
```

Comment:
The event handler function receives ```e``` as a parameter, so it's not required that you pass it as an anonymous function. 

I'm not sure which one is the best practice 🤔. Whether it is passing the handler as anonymous or name it, I just wanted to tell you that in this case you can pass the named function like ```onClick={handleClick}``` and it will work just fine 😸